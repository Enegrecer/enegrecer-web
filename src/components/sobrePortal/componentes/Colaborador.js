import React from 'react';
import PropTypes from 'prop-types';

const Colaborador = props => (
  <div className="col s12 m4 person">
    <div className="conteudo">
      <img alt={`Foto de ${props.nome}`} src={props.foto} />
      <div>
        <h5>{ props.nome }</h5>
        <p>{ props.papel }</p>
      </div>
    </div>
  </div>
);

Colaborador.propTypes = {
  nome: PropTypes.string.isRequired,
  papel: PropTypes.string.isRequired,
  foto: PropTypes.string.isRequired,
};

export default Colaborador;
