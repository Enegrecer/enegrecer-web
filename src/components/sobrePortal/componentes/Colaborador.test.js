import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Colaborador from './Colaborador';

describe('Colaborador', () => {
  const propsParaColaborador = {
    nome: 'Pessoa Colaboradora',
    papel: 'Trabalhadora',
    foto: 'http://google.com',
  };

  const wrapper = shallow(<Colaborador {...propsParaColaborador} />);

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
