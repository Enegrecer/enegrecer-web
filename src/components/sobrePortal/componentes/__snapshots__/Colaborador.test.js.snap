// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colaborador possui a mesma estrutura 1`] = `
<div
  className="col s12 m4 person"
>
  <div
    className="conteudo"
  >
    <img
      alt="Foto de Pessoa Colaboradora"
      src="http://google.com"
    />
    <div>
      <h5>
        Pessoa Colaboradora
      </h5>
      <p>
        Trabalhadora
      </p>
    </div>
  </div>
</div>
`;
